core/src/main/java/com/novoda/landingstrip/LandingStripAdapter.java

```diff
@@ -0,0 +1,38 @@
+package com.novoda.landingstrip;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+public abstract class LandingStripAdapter<T extends View> {
+
+    private Listener listener;
+
+    protected abstract T createView(ViewGroup parent, int position);
+
+    protected abstract void bindView(T view, int position);
+
+    void setListener(Listener listener) {
+        this.listener = listener;
+    }
+
+    public void notifyDataSetChanged() {
+        if (listener != null) {
+            listener.onNotifyDataSetChanged(this);
+        }
+    }
+
+    public void notifyItemChanged(int position) {
+        if (listener != null) {
+            listener.onNotifyItemChanged(this, position);
+        }
+    }
+
+    protected abstract int getCount();
+
+    interface Listener {
+
+        <T extends View> void onNotifyDataSetChanged(LandingStripAdapter<T> adapter);
```

Comment:
Because `T` is a client concept, the library doesn't know T at the point of creating the `Listener` https://github.com/novoda/landing-strip/pull/32/files/0d559730914d7e4f1e3278979c5845fe1d6bdb05#diff-03f70e68af4374fc21ddd30d94ca1309R57

`T` is only known once `setAdapter` has been called.

Perhaps it would be better to create the `listener<T>` in `setAdapter` instead of allowing dynamic types